all_test.go

```diff
@@ -500,4 +534,80 @@ func TestSessionDelete(t *testing.T) {
 	if err != ErrInvalidSessionToken {
 		t.Error("Expected ErrInvalidSessionToken after Logout")
 	}
+
+	teardown(c)
+}
+
+func TestUpdateIdentity(t *testing.T) {
+	c := setup1(t)
+
+	newEmail := "newEmail"
+	newUsername := "newUsername"
+	newName := "newName"
+	newSurname := "newSurname"
+	newMobile := "newMobile"
+
+	if err := c.UpdateIdentity(notFoundUserId, newEmail, newUsername, newName, newSurname, newMobile); err != nil {
+		if err != ErrIdentityAuthNotFound {
+			t.Fatalf("TestUpdateIdentity failed: an error occurred: %v", err)
+		}
+	} else {
+		t.Fatalf("Update should not have succeeded")
+	}
+
+	if err := c.UpdateIdentity(joeUserId, newEmail, newUsername, newName, newSurname, newMobile); err != nil {
+		t.Fatalf("Update should not have failed")
+	}
+
+	users, e := c.GetAuthenticatorIdentities()
+	if e != nil {
+		t.Fatalf("TestUpdateIdentity failed: $v", e)
+	}
+
+	updateSuccess := false
+	for _, user := range users {
+		if user.UserId == joeUserId && CanonicalizeIdentity(user.Email) == CanonicalizeIdentity(newEmail) && user.Username == newUsername && user.Firstname == newName && user.Lastname == newSurname && user.Mobilenumber == newMobile {
+			updateSuccess = true
+			break
+		}
+	}
+	if !updateSuccess {
+		t.Fatalf("TestUpdateIdentity failed")
```

Comment:
It would be better here have an information error message such as "After update, failed to find updated user"